src/app/lib/button/button.component.ts

```diff
@@ -1,16 +1,42 @@
-import { Component, EventEmitter, Output, Input } from '@angular/core';
-import { ButtonOptions } from './button-options';
+/* tslint:disable:component-selector */
+// https://github.com/mgechev/codelyzer/issues/178#issuecomment-265154480
+
+import {Component, ElementRef, Input, OnChanges, Renderer2, SimpleChanges} from '@angular/core';
+
+export function throwErrorForInvalidButtonColor() {
+    throw new Error('Unsupported color input value');
+}
+
+type ButtonColor = 'primary' | 'primary-alt1' | 'primary-alt2' | 'primary-alt3' | 'secondary' | 'tertiary';
 
 @Component({
-  selector: 'hc-button',
-  templateUrl: './button.component.html',
-  styleUrls: ['./button.component.scss']
+    selector: 'button[hc-button], a[hc-button]',
+    template: '<ng-content></ng-content>',
+    styleUrls: ['./button.component.scss']
 })
-export class ButtonComponent {
-  @Input() public options = new ButtonOptions();
-  @Output() public onButtonClick = new EventEmitter<boolean>();
```

Comment:
This is a nice change that the consumer can use the native `(onClick)` functionality rather than a pass through EventEmitter